harness/harness.go

```diff
@@ -61,13 +61,16 @@ func (h *Harness) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Flush any change events and rebuild app if necessary.
-	// Render an error page if the rebuild / restart failed.
-	err := watcher.Notify()
-	if err != nil {
-		atomic.CompareAndSwapInt32(&lastRequestHadError, 0, 1)
-		renderError(w, r, err)
-		return
+	// If app did not start when harness was run then trigger the build to capture the error
+	if h.app==nil {
+		// Flush any change events and rebuild app if necessary.
+		// Render an error page if the rebuild / restart failed.
+		err := h.watcher.Notify()
+		if err != nil {
+			atomic.CompareAndSwapInt32(&lastRequestHadError, 0, 1)
```

Comment:
You are correct, I guess there could be a race condition where the browser requests the favicon immediately following the page request. I don't know if it is crucial to be this way - but it isnt hurting anything by being there